core/src/main/java/org/apache/gravitino/authorization/OwnerEventManager.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.authorization;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Optional;
+import lombok.Getter;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.listener.EventBus;
+import org.apache.gravitino.listener.api.event.GetOwnerPreEvent;
+import org.apache.gravitino.listener.api.event.SetOwnerEvent;
+import org.apache.gravitino.listener.api.event.SetOwnerFailureEvent;
+import org.apache.gravitino.listener.api.event.SetOwnerPreEvent;
+import org.apache.gravitino.listener.api.info.OwnerInfo;
+import org.apache.gravitino.utils.MetadataObjectUtil;
+import org.apache.gravitino.utils.PrincipalUtils;
+
+public class OwnerEventManager extends OwnerManager {
```

Comment:
Could you use `OwnerDispatcher` like other event dispatcher like `AccessControlDispatcher`?

Reply:
OwnerManager is not a interface. Should we need to introduce the owner operations dispatcher? What's your opinion, @jerqi . I have no preference.

Reply:
It may be better to extract an interface

Reply:
done